### PR TITLE
pyupgrade --py38-plus

### LINKED ISSUE #662: SyntaxWarning: Invalid escape sequence '\d'
Unknown:

Cement Framework 3.0.9
Python 3.12.1
Platform Linux-6.4.16-linuxkit-x86_64-with

```
|> cement-py312 <| src # make test
flake8 cement/ tests/
<unknown>:81: SyntaxWarning: invalid escape sequence '\d'
```

### PR DESCRIPTION
**Pull request by sigma67**
**NOTICE:** Prior to submitting a pull request, please ensure you abide by the
[Guidelines for Code Contributions], including but not limited to:

- PR is associated with at least one issue
- Submit from a **topic** branch, not a pre-existing Cement branch
- Make every effort to comply with [PEP8]
- All tests pass successfully
- Coverage reports 100% code coverage

---

**Issue:** Fixes #662. Additionally a cleanup effort.

All changes are the result of

```bash
pyupgrade --py38-plus cement/**/*.py
pyupgrade --py38-plus tests/**/*.py
```

[Guidelines for Code Contributions]: https://github.com/datafolklabs/cement/blob/master/.github/CONTRIBUTING.md#guidelines-for-code-contributions
[PEP8]: http://www.python.org/dev/peps/pep-0008/

### CODE REVIEW

**Comment by derks** [comment]
@sigma67 this is way too big a change to pull into one PR (for me). Also not in love with some of the changes (like removing the first empty line of every file... personal pref I guess).  There's also a lot of changes in `cli/contrib` which I would not want to make as those modules should remain in line with upstream.  

Some thoughts:

- Good call-out that I need to ensure anything in `cli/contrib` is up-to-date with upstream... but would not want this "covered" by `pyupgrade` (unless something broke)... on that note maybe should consider dropping contrib and adding meta package cement[cli] like we've done with extensions. hrm...
- Does `pyupgrade` have some sort of "report" capability? I'd much rather add this as a secondary to `make comply` (flake8) that we can start to run and work _toward_ cleanup and not attempt to do it all in one go.  Something like Coverage's HTML report would be fantastic.
- Once cleanup is complete, then we could add the pre-check to `make comply` which could then block `make test`... enforcing py38+ going forward.

In summary (today):

Could we look at adding a check/report that can be run as `make comply-py38+` as a starting point?

**Comment by sigma67** [comment]
I added the make command `make comply-py` (will not remain 38 forever). pyupgrade always rewrites unfortunately, so you should only run on a clean git status. You can easily reset with a `git checkout HEAD -- cement`.

I ran pyupgrade only on the tests directory through pyupgrade for now to reduce the changeset

**Comment by sigma67** [comment]
Also regarding the empty lines at the top of the module -- pep8 actually mentions a module docstring which should be placed there instead. 

https://pep8.org/#module-level-dunder-names

So could be a good idea to take this opportunity to replace the empty line with a module docstring

**Comment by derks** [comment]
@sigma67 I was hoping to find a `--check` or `--dry-run` option for the pyupgrade utility... as I greatly prefer being warned of something, and then controlling how/what is changed. This feature does not exist by design of the maintainer, and based on their response to the subject I will not be merging this change in.

- Ref: https://github.com/asottile/pyupgrade/issues/442

Ruff claims to replace pyupgrade... perhaps that may be an alternative route to solve this (and potentially replace flake8 at same time)?

- https://pypi.org/project/ruff/

**Comment by sigma67** [comment]
Sounds great, with ruff you can disable individual pyupgrade rules easily.

I'll reduce the PR to a fix for #662 and then get started on ruff
